middlewares/accesslog/logger.go

```diff
@@ -66,7 +67,18 @@ func NewLogHandler(config *types.AccessLog) (*LogHandler, error) {
 		Hooks:     make(logrus.LevelHooks),
 		Level:     logrus.InfoLevel,
 	}
-	return &LogHandler{logger: logger, file: file, filePath: config.FilePath}, nil
+
+	headerredactions := make(map[string]bool)
```

Comment:
instead of a `map[string]bool` prefer `map[string]struct`

```go
for _, header := range config.HeaderRedactions {
	headerredactions[header] = struct{}{}
}
```

https://dave.cheney.net/2014/03/25/the-empty-struct

Reply:
Personally I prefer `bool` as it seems less obscure compared to a `struct`, especially with regards to a set: an element is either part of a set or not.

Reply:
empty struct is a zero memory allocation tricks.